File: app/src/main/java/com/thefuturestic/digitalservices/GraphicCategory/GraphicCategoryActivity.java
package com.thefuturestic.digitalservices.GraphicCategory;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.StaggeredGridLayoutManager;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;

import com.thefuturestic.digitalservices.R;

public class GraphicCategoryActivity extends AppCompatActivity {
    RecyclerView recyclerView;
    private ImageView back;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_graphic_ccat);

        recyclerView = (RecyclerView) findViewById(R.id.rvMain);
        back = (ImageView)findViewById(R.id.back_btn);
        Bitmap[] bitmaps = setUpBitmaps();
        recyclerView.setAdapter(new GraphicCategoryGridLayoutAdapter(bitmaps));
        StaggeredGridLayoutManager mLayoutManager = new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(mLayoutManager);

        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
    }

    private Bitmap[] setUpBitmaps() {
        Bitmap[] bitmaps = new Bitmap[8];

        bitmaps[0] = BitmapFactory.decodeResource(getResources(), R.drawable.bigger_one);
        bitmaps[1] = BitmapFactory.decodeResource(getResources(), R.drawable.samll_one);
        bitmaps[2] = BitmapFactory.decodeResource(getResources(), R.drawable.bigger_two);
        bitmaps[3] = BitmapFactory.decodeResource(getResources(), R.drawable.small_two);
        bitmaps[4] = BitmapFactory.decodeResource(getResources(), R.drawable.bigger_three);
        bitmaps[5] = BitmapFactory.decodeResource(getResources(), R.drawable.small_three);
        bitmaps[6] = BitmapFactory.decodeResource(getResources(), R.drawable.bigger_four);
        bitmaps[7] = BitmapFactory.decodeResource(getResources(), R.drawable.small_four);



        return bitmaps;
    }
}

File: app/src/main/java/com/thefuturestic/digitalservices/GraphicCategory/GraphicCategoryActivity.kt
package com.thefuturestic.digitalservices.GraphicCategory

import android.os.Bundle
import android.widget.ProgressBar
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.database.DatabaseReference
import com.thefuturestic.digitalservices.R


class GraphicCategoryActivity : AppCompatActivity() {
    private var progressBar: ProgressBar? = null
    internal lateinit var mRecyclerView: RecyclerView
    internal lateinit var mRef: DatabaseReference
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_category)

//
//        val inn = Intent()
//        val categoryTitle = inn.extras!!.getString("categoryTitle")
//        if(categoryTitle.isNullOrEmpty()){
//            val i = Intent(this, Home::class.java)
//           startActivity(i)
//            finish()
//        }else{
//            mRecyclerView = findViewById(R.id.recyclerView_home)
//            mRecyclerView.setHasFixedSize(true)
//            mRecyclerView.layoutManager = LinearLayoutManager(this)
//            mRecyclerView.layoutManager = LinearLayoutManager(this)
//
//            mRef = FirebaseDatabase.getInstance().getReference(""+categoryTitle)
//            mRef!!.keepSynced(true)
//
//            //progress bar
//            progressBar = findViewById(R.id.progress_bar_home)
//            progressBar!!.setVisibility(View.VISIBLE)
//        }

    }

//    override fun onStart() {
//        progressBar!!.setVisibility(View.VISIBLE)
//        super.onStart()
//        val firebaseRecyclerAdapter = object : FirebaseRecyclerAdapter<ModelCategory, ViewHolderCategory>(
//            ModelCategory::class.java,
//            R.layout.item_home_view,
//            ViewHolderCategory::class.java,
//
//            mRef
//        ) {
//            override fun populateViewHolder(viewHolder: ViewHolderCategory, model: ModelCategory, position: Int) {
//                // viewHolder.setDetails(context!!, model.img)
//                model.img?.let {
//                    viewHolder.setDetails(this@CategoryActivity, model.name,model.categoryTitle,
//                        it
//                    )
//                }
//
//                progressBar!!.setVisibility(View.GONE)
//            }
//
//        }
//        mRecyclerView!!.adapter = firebaseRecyclerAdapter
//    }
}
